src/engine/QueryPlanner.cpp

```diff
@@ -2461,16 +2465,18 @@ auto QueryPlanner::createJoinWithTransitivePath(const SubtreePlan& a,
     return std::nullopt;
   }
 
+  // Do not bind the side of a path twice and don't bind on graph variable.
   auto joinCols = getJoinColumnsForTransitivePath(jcs, aTransPath != nullptr);
-  // Do not bind the side of a path twice and don't bind on graph variable
   if (!joinCols.has_value()) {
     return std::nullopt;
   }
-  const auto& [thisCol, otherCol] = joinCols.value();
+
   // An unbound transitive path has at most two columns we can bind to.
+  const auto& [thisCol, otherCol] = joinCols.value();
   AD_CONTRACT_CHECK(thisCol <= 1);
-  // The left or right side is a TRANSITIVE_PATH and its join column
-  // corresponds to the left side of its input.
+
+  // The left or right side is a transitive path and its join column corresponds
+  // to the left side of its input.
```

Comment:
This comment was already confusing before this change. If `thisCol == 0` we bind the left side, if `thisCol == 1`, we bind the right side. That simple (other values for `thisCol` are not allowed here).